crates/core_arch/src/riscv/mod.rs

```diff
@@ -0,0 +1,10 @@
+//! RISC-V intrinsics
+
+/// Generates the `PAUSE` instruction
+///
+/// The PAUSE instruction is a HINT that indicates the current hart's rate of instruction retirement
+/// should be temporarily reduced or paused. The duration of its effect must be bounded and may be zero.
+#[inline(always)]
+pub unsafe fn pause() {
```

Comment:
You should add a feature that maps to the extension, see what it is done for the other arch extensions.

Reply:
Yes, adding feature gate to extension specific instructions are necessary. For pause instruction, this instruction is in `HINT` space (in `I` set which must be implemented by RISC-V cores), whether there is Zihintpause or not does not change architectural state after instruction is retired. It will always run on all RISC-V cores. I don't know if it's still required to add feature gate to this function, or should we write it down in documentation for clarification. :)

Reply:
You may add a comment about it.

I assume nothing else would use the same value so either it is ignored or it is used as intended and would not change in unexpected ways the behavior in the future.

Can you confirm that I understood correctly?